### ISSUE TITLE
RedisGraph::Metadata.property_keys does not properly invalidate

### ISSUE THREAD

**Issue by jmsanders** [RedisGraph/redisgraph-rb#6]
`SET` queries don't return the correct property keys:

```rb
irb(main):001:0> require "redisgraph"
=> true
irb(main):002:0> redis = RedisGraph.new("demo")
irb(main):003:0> redis.query("MATCH (n) RETURN (n)").resultset
=> []
irb(main):004:0> redis.query("CREATE (n {foo: 1}) RETURN (n)").resultset
=> [[[{"foo"=>1}]]]
irb(main):005:0> redis.query("MATCH (n {foo: 1}) SET n.bar = 2 RETURN (n)").resultset
=> [[[{"foo"=>1}, {nil=>2}]]]
```

Instead of returning a node with properties `"foo"` and `"bar"`, it returns a node with properties `"foo"` and `nil`.

I can still properly interact with the property:

```rb
irb(main):006:0> redis.query("MATCH (n {foo: 1}) SET n.bar = 3 RETURN (n)").resultset
=> [[[{"foo"=>1}, {nil=>3}]]]
```

And if I interact with Redis via a different client (such as redis-cli), the property key is indeed set:

```sh
127.0.0.1:6379> GRAPH.QUERY demo "MATCH (n) RETURN (n)"
1) 1) "n"
2) 1) 1) 1) 1) "id"
            2) (integer) 0
         2) 1) "labels"
            2) (empty array)
         3) 1) "properties"
            2) 1) 1) "foo"
                  2) (integer) 1
               2) 1) "bar"
                  2) (integer) 3
3) 1) "Query internal execution time: 0.216800 milliseconds"
```

I believe this is because `RedisGraph::Metadata.property_keys` is being memoized:

https://github.com/RedisGraph/redisgraph-rb/blob/daf7c70bd2707b2e4dd20de7e9bfbb634188cece/lib/redisgraph.rb#L34

The initial call is indeed calling the proc but otherwise, I don't believe this comment is accurate:

https://github.com/RedisGraph/redisgraph-rb/blob/daf7c70bd2707b2e4dd20de7e9bfbb634188cece/lib/redisgraph.rb#L18-L19

Manually invalidating `RedisGraph::Metadata` doesn't seem fix my issue:

```rb
irb(main):007:0> redis.metadata.invalidate
=> nil
irb(main):008:0> redis.query("MATCH (n) RETURN (n)").resultset
=> [[[{"foo"=>1}, {nil=>3}]]]
```

Initializing a completely new connection does:

```rb
irb(main):009:0> redis = RedisGraph.new("demo")
irb(main):010:0> redis.query("MATCH (n) RETURN (n)").resultset
=> [[[{"foo"=>1}, {"bar"=>3}]]]
```

Removing the memoization and replacing it with a call to the proc each time would fix the particular issue I'm seeing, but I suspect that comes at additional performance tradeoffs so I'm opening this issue to bring it to your attention instead of PRing a fix. Thanks!

**Comment by jeffreylovitz** [comment]
@jmsanders Thanks for catching this! You're correct about the performance tradeoff; I will write a PR that handles this more robustly shortly.

**Comment by jeffreylovitz** [comment]
Hi @jmsanders ,

I've resolved this bug and released a new gem - v2.0.1

Let me know if you encounter further issues!

**Comment by jmsanders** [comment]
Those fixes look good to me - I'll go ahead and close for now. Thanks for the quick turnaround!
